src/Features/CSharp/Portable/UseIsNullCheck/CSharpUseIsNullCheckForCastAndEqualityOperatorCodeFixProvider.cs

```diff
@@ -0,0 +1,92 @@
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseIsNullCheck
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    internal class CSharpUseIsNullCheckForCastAndEqualityOperatorCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(IDEDiagnosticIds.UseIsNullCheckForCastAndEqualityOperatorDiagnosticId);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics.First();
+
+            context.RegisterCodeFix(
+                new MyCodeAction(CSharpFeaturesResources.Use_is_null_check,
+                c => this.FixAsync(context.Document, diagnostic, c)),
+                context.Diagnostics);
+            return SpecializedTasks.EmptyTask;
+        }
+
+        protected override Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+
+            foreach (var diagnostic in diagnostics)
+            {
+                var binary = (BinaryExpressionSyntax)diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken: cancellationToken);
+                var rewritten = Rewrite(binary);
+
+                editor.ReplaceNode(binary, rewritten);
+            }
+
+            return SpecializedTasks.EmptyTask;
+        }
+
+        private ExpressionSyntax Rewrite(BinaryExpressionSyntax binary)
+        {
+            var isPattern = RewriteWorker(binary);
+            if (binary.IsKind(SyntaxKind.EqualsExpression))
+            {
+                return isPattern;
+            }
+
+            // convert:  (object)expr != null   to    !(expr is null)
+            return SyntaxFactory.PrefixUnaryExpression(
+                SyntaxKind.LogicalNotExpression,
+                SyntaxFactory.ParenthesizedExpression(isPattern.WithoutTrivia())).WithTriviaFrom(isPattern);
+        }
+
+        private IsPatternExpressionSyntax RewriteWorker(BinaryExpressionSyntax binary)
+            => binary.Right.IsKind(SyntaxKind.NullLiteralExpression)
+                ? Rewrite(binary, binary.Left, binary.Right)
+                : Rewrite(binary, binary.Right, binary.Left);
+
+        private IsPatternExpressionSyntax Rewrite(
+            BinaryExpressionSyntax binary, ExpressionSyntax expr, ExpressionSyntax nullLiteral)
+        {
+            var castExpr = (CastExpressionSyntax)expr;
+            return SyntaxFactory.IsPatternExpression(
+                castExpr.Expression.WithTriviaFrom(binary.Left),
```

Comment:
`WithTriviaFrom(binary.Left)`? should this be `expr` or `castExpr` instead?

Reply:
if you do that, consider removing the `binary` argument and just passing `operatorToken`

Reply:
Test added.